Change `HelpTheme` type to allow sub-classing

In Click Extra, I have a [default theme lying around](https://github.com/kdeldycke/click-extra/blob/5245748ef30f0a2fba2a7899530c203445907c49/click_extra/colorize.py#L48) that try to extends `cloup.HelpTheme` with additional categories of styles. Like [log-level colors](https://github.com/kdeldycke/click-extra/blob/5245748ef30f0a2fba2a7899530c203445907c49/click_extra/colorize.py#L75-L80) and [other help screen details](https://github.com/kdeldycke/click-extra/blob/5245748ef30f0a2fba2a7899530c203445907c49/click_extra/colorize.py#L82-L92).

The problem is that `cloup.HelpTheme` is a `NamedTuple`. And I cannot easily sub-class it. [As you can see in my code](https://github.com/kdeldycke/click-extra/blob/5245748ef30f0a2fba2a7899530c203445907c49/click_extra/colorize.py#L48), I had to redefine the whole properties and methods.

I also encountered issues with [MyPy which cannot solve references to sub-styles](https://github.com/kdeldycke/click-extra/blob/5245748ef30f0a2fba2a7899530c203445907c49/click_extra/version.py#L66-L67):
```pytb
click_extra/version.py:58: error: Cannot determine type of "default_theme"  [has-type]
        default_package_name_style: IStyle | None = default_theme.invoked_command
                                                    ^~~~~~~~~~~~~
click_extra/version.py:59: error: Cannot determine type of "default_theme"  [has-type]
        default_prog_name_style: IStyle | None = default_theme.invoked_command
                                                 ^~~~~~~~~~~~~
```

That is somewhat expected, as [`NamedTuple` subclasses is flagged as `wontfix` by Python's `typing` project](https://github.com/python/typing/issues/427).

@janluke, would you be open to change the type of `HelpTheme` so I can reuse it seamlessly?

I naively think that a class is the natural choice, but maybe a `@dataclass` might be another alternative to `NamedTuple`. I'm pretty sure you have a better idea of what `HelpTheme` should look like in that case.

--- a/cloup/styling.py
+++ b/cloup/styling.py
@@ -2,8 +2,10 @@
 This module contains components that specifically address the styling and theming
 of the ``--help`` output.
 """
+import dataclasses
 import dataclasses as dc
-from typing import Any, Callable, Dict, NamedTuple, Optional
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional
 
 import click
 
@@ -14,8 +16,8 @@ IStyle = Callable[[str], str]
 """A callable that takes a string and returns a styled version of it."""
 
 
-# noinspection PyUnresolvedReferences
-class HelpTheme(NamedTuple):
+@dataclass(frozen=True)
+class HelpTheme:
     """A collection of styles for several elements of the help page.
 
     A "style" is just a function or a callable that takes a string and returns
@@ -95,7 +97,7 @@ class HelpTheme(NamedTuple):
             del kwargs["alias_secondary"]
         kwargs.pop('self')
         if kwargs:
-            return self._replace(**kwargs)
+            return dataclasses.replace(self, **kwargs)
         return self
 
     @staticmethod

